### PR TITLE
NNEO-10927 subscription api endpoints

### PR DESCRIPTION
**Pull request by zbigniewwiech**

### CODE REVIEW

**Review by jgivoni — CHANGES_REQUESTED** [review]

File: src/Gateway/SubscriptionBase.php [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+use VendoSdk\Gateway\Request\Details\ExternalReferences;
+use VendoSdk\Gateway\Request\Details\Item;
+use VendoSdk\Gateway\Request\Details\Request;
+use VendoSdk\Gateway\Request\Details\ShippingAddress;
+use VendoSdk\Gateway\Response\PaymentResponse;
+use VendoSdk\Gateway\Response\SubscriptionResponse;
+use VendoSdk\Util\HttpClientTrait;
+use VendoSdk\Vendo;
+
+abstract class SubscriptionBase
```
[the remark this answers is not in the document]
Comment (zbigniewwiech) [reply src/Gateway/SubscriptionBase.php:18]: different response, with "subscription_id"

File: examples/gateway/credit_card_signup.php [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        echo "\n   1. Save the payment token: " . $response->getPaymentToken();
+        echo "\n   2. Redirect the user to the verification URL: " . $response->getResultDetails()->getVerificationUrl();
+        echo "\nthe user will verify his payment details, then he will be redirected to the Success URL that's configured in your account at Vendo's back office.";
+        echo "\nwhen the user comes back you need to post the request to vendo again, you can use the TokenPayment class.";
+    }
+    echo "\n\n\n";
+
+
+} catch (\VendoSdk\Exception $exception) {
+var_dump($exception->getMessage());die;
```
[the remark this answers is not in the document]
Comment (zbigniewwiech) [reply examples/gateway/credit_card_signup.php:129]: oopssss!!! :(

[the remark this answers is not in the document]
Comment (zbigniewwiech) [reply examples/gateway/credit_card_signup.php]: removed

File: src/Gateway/SubscriptionBase.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (jgivoni) [reply src/Gateway/SubscriptionBase.php:18]: sorry, got confused, closing

**Review by jgivoni — APPROVED** [review]

**Pull request merged**